AlicerceTests/ServiceLocatorTests.swift

```diff
@@ -0,0 +1,301 @@
+            
+        } catch let error {
+            XCTAssert(false, "💥: Register failed with error: \(error)")
+        }
+    }
+    
+    func testLazyInitRegisterUsingTypeInference() {
+        
+        let testLazyServiceString: () -> (String) = {
+            return "Test Register Service"
+        }
+        let testLazyServiceDouble: () -> (Double) = {
+            return 1.0
+        }
+        
+        do {
+            
+            // Register services
+            try serviceLocator.register(testLazyServiceString)
+            try serviceLocator.register(testLazyServiceDouble)
+            
+            // Get registered services
+            let rStringService: String = try serviceLocator.get()
+            let rDoubleService: Double = try serviceLocator.get()
+            
+            // Validate registered services
+            XCTAssertEqual(rStringService, testLazyServiceString())
+            XCTAssertEqual(rDoubleService, testLazyServiceDouble())
+            
+        } catch let error {
+            XCTAssert(false, "💥: Register failed with error: \(error)")
+        }
+    }
+    
+    func testLazyInitRegisterUsingName() {
+        
+        let testLazyServiceString: () -> (String) = {
+            return "Test Register Service"
+        }
+        let serviceName = "TestServiceName"
+        
+        do {
+            
+            // Register services
+            try serviceLocator.register(name: serviceName, testLazyServiceString)
+            
+            // Get registered services
+            let rStringService: String = try serviceLocator.get(name: serviceName)
+            
+            // Validate registered services
+            XCTAssertEqual(rStringService, testLazyServiceString())
+            
+        } catch let error {
+            XCTAssert(false, "💥: Register failed with error: \(error)")
+        }
+    }
+    
+    func testRegisterUsingServiceTypeInference_GetUsingName() {
+        
+        typealias DoubleService = Double
+        
+        let testServiceString = "Test Register Service"
+        let testServiceDouble = 1.0
+        
+        do {
+            
+            // Register services
+            try serviceLocator.register(service: testServiceString)
+            try serviceLocator.register(service: testServiceDouble)
+            
+            
+            // Get registered services
+            let rStringService: String = try serviceLocator.get(name: "String")
+            let rDoubleService: Double = try serviceLocator.get(name: "Double")
+            
+            
+            // Validate registered services
+            XCTAssertEqual(rStringService, testServiceString)
+            XCTAssertEqual(rDoubleService, testServiceDouble)
+            
+            
+        } catch let error {
+            XCTAssert(false, "💥: Register failed with error: \(error)")
+        }
+    }
+    
+    func testRegisterUsingName_GetUsingTypeInference() {
```

Comment:
sure, i can remove them but if we return the name when we register, we can do things like this